src/main/java/ro/utcn/sd/agui/a1/persistence/jdbc/JdbcQuestionRepository.java

```diff
@@ -0,0 +1,120 @@
+package ro.utcn.sd.agui.a1.persistence.jdbc;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import ro.utcn.sd.agui.a1.entity.Question;
+import ro.utcn.sd.agui.a1.entity.Tag;
+import ro.utcn.sd.agui.a1.persistence.QuestionRepository;
+
+import java.util.*;
+
+@RequiredArgsConstructor
+public class JdbcQuestionRepository implements QuestionRepository {
+
+    private final JdbcTemplate template;
+
+    @Override
+    public Question save(Question question) {
+        if(question.getQuestionId()!=null){
+            update(question);
+        }
+        else
+        {
+            int id = insert(question);
+            question.setQuestionId(id);
+        }
+
+
+
+        return question;
+    }
+
+    @Override
+    public Optional<Question> findById(int id) {
+        List<Question> questions = template.query("SELECT * FROM question WHERE question_id = ?",
+                new Object[]{id},
+                (resultSet, i) -> new Question(
```

Comment:
There is some duplicated code here: you could extract this lambda into a dedicated `QuestionMapper` class.